src/Tribe/JSON_LD/Abstract.php

```diff
@@ -52,34 +54,27 @@ public static function instance( $name = null ) {
 	 *
 	 * @param mixed $post Either a post ID or a WP_post object.
 	 * @param array $args {
-	 *       Optional. An array of arguments to control the returned data.
+	 *      Optional. An array of arguments to control the returned data.
 	 *
 	 *      @type string $context         The value of the `@context` tag, defaults to 'https://schema.org'
-	 *      @type bool   $skip_duplicates If set to `true` fetching data for same post a second time will
-	 *                                    return an empty array. Default `true`.
 	 * }
 	 *
 	 * @return array Either an array containing a post data or an empty array if the post data cannot
 	 *               be generated, the `$post` parameter is not a valid post ID or object or the data
-	 *               for the post has been fetched already and the `skip_duplicates` argument is truthy.
+	 *               for the post has been fetched already.
 	 */
 	public function get_data( $post = null, $args = array() ) {
-		$post = $this->get_post_object( $post );
-
-		if ( empty( $post ) ) {
+		// This prevents a JSON_LD from existing twice one the same page
+		if ( $this->exists( $post ) ) {
```

Comment:
Should this check go below the `! $post_id` if statement? Then you can call `$this->exists( $post_id )` which would be ever so slightly less expensive.